app/src/main/java/app/revanced/manager/ui/viewmodel/MainViewModel.kt

```diff
@@ -38,7 +82,39 @@ class MainViewModel(
         }
     }
 
-    fun applyLegacySettings(data: String) = viewModelScope.launch {
+    fun importLegacySettings(componentActivity: ComponentActivity) {
+        if (!prefs.firstLaunch.getBlocking()) return
+
+        try {
+            val launcher = componentActivity.registerForActivityResult(
+                ActivityResultContracts.StartActivityForResult()
+            ) { result: ActivityResult ->
+                if (result.resultCode == ComponentActivity.RESULT_OK) {
+                    result.data?.getStringExtra("data")?.let {
+                        applyLegacySettings(it)
+                    } ?: app.toast(app.getString(R.string.legacy_import_failed))
+                } else {
+                    app.toast(app.getString(R.string.legacy_import_failed))
+                }
+            }
+
+            val intent = Intent().apply {
+                setClassName(
+                    "app.revanced.manager.flutter",
+                    "app.revanced.manager.flutter.ExportSettingsActivity"
+                )
+            }
+
+            launcher.launch(intent)
+        } catch (e: Exception) {
+            if (e !is ActivityNotFoundException) {
+                app.toast(app.getString(R.string.legacy_import_failed))
+                Log.e(tag, "Failed to launch legacy import activity: $e")
+            }
+        }
+    }
+
+    private fun applyLegacySettings(data: String) = viewModelScope.launch {
```

Comment:
The final line of this function sets `firstLaunch` to false, but we also made the auto updates dialog mandatory even if importing succeeds.

I think we need to remove the final line to show the dialog if importing succeeds.